forest/cli/src/cli/snapshot_cmd.rs

```diff
@@ -0,0 +1,326 @@
+                    .unwrap_or_else(|| default_snapshot_dir(&config));
+                match snapshot_fetch(&snapshot_dir, config).await {
+                    Ok(out) => println!("Snapshot successfully downloaded at {}", out.display()),
+                    Err(e) => cli_error_and_die(format!("Failed fetching the snapshot: {e}"), 1),
+                }
+            }
+            Self::Dir => {
+                let dir = default_snapshot_dir(&config);
+                println!("Default snapshot dir: {:?}", dir);
+            }
+            Self::List { snapshot_dir } => {
+                let snapshot_dir = snapshot_dir
+                    .clone()
+                    .unwrap_or_else(|| default_snapshot_dir(&config));
+                println!("Snapshot dir: {:?}", snapshot_dir);
+                if let Ok(dir) = fs::read_dir(snapshot_dir) {
+                    println!("\nLocal snapshots:");
+                    for entry in dir.flatten() {
+                        let path = entry.path();
+                        if path.is_file() {
+                            if let Some(filename) = entry.file_name().to_str() {
+                                if filename.ends_with(".car") {
+                                    println!("{filename}");
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            Self::Remove {
+                filename,
+                snapshot_dir,
+                yes,
+            } => {
+                let yes = *yes;
+                let snapshot_dir = snapshot_dir
+                    .clone()
+                    .unwrap_or_else(|| default_snapshot_dir(&config));
+                let mut snapshot_path = snapshot_dir;
+                snapshot_path.push(filename);
+                if snapshot_path.exists()
+                    && snapshot_path.is_file()
+                    && snapshot_path.extension() == Some(OsStr::new("car"))
+                {
+                    let term = Term::stdout();
+                    term.write_line(&format!("Deleting {:?}", snapshot_path))
+                        .unwrap();
+                    if !yes
+                        && !Confirm::new()
+                            .with_prompt("Do you want to continue?")
+                            .interact()
+                            .unwrap()
+                    {
+                        term.write_line("Aborted.").unwrap();
+                        return;
+                    }
+
+                    delete_snapshot(&snapshot_path);
+                } else {
+                    let term = Term::stdout();
+                    term.write_line(&format!(
+                        "{:?} is not a valid snapshot file path, to list all snapshots, run forest-cli snapshot list",
+                        snapshot_path)
+                    ).unwrap();
+                }
+            }
+            Self::Prune { snapshot_dir, yes } => {
```

Comment:
The same comment for `List`, having 10 levels of indentation begs for refactoring. Also, there seem to be a lot of opportunity for `map`, `filter` and `filter_map`.

Reply:
fixed